File: 0050-powx-n/0050-powx-n.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double ans = 1;
        long ng = n;
        if(ng<0){
            ng = -1 * ng;
        }
        while(ng>0){
            if(ng%2==1){
                ans = ans * x;
                ng = ng - 1;
            }
            else{
                x=x*x;
                ng = ng/2;
            }
        }
        if(n<0) ans = (double)(1.0)/(double)(ans);
        return ans;
    }
};

File: 0050-powx-n/0050-powx-n.java
class Solution {
    public double myPow(double x, int n) {
        double ans =1;
        // if(n==Integer.MIN_VALUE) return 0;
        if(n==Integer.MIN_VALUE){ 
            if(x==1 || x==-1){ return 1;}
            return 0;
        }
       
        boolean m = true;
        if(n<0){ m= false; n=-n;}
        
        while(n>0){
            if(n%2==1){
                ans=ans*x;
                n=n-1;
            }else{
                n=n/2;
                x=x*x;
            }
        }
        if(m){ return ans;}
        else{ return 1.0/ans;}

    }
}